casper/src/main/scala/coop/rchain/casper/blocks/proposer/ProposeResult.scala

```diff
@@ -24,7 +24,10 @@ case object NotEnoughNewBlocks         extends CheckProposeConstraintsFailure
 case object TooFarAheadOfLastFinalized extends CheckProposeConstraintsFailure
 
 object CheckProposeConstraintsResult {
-  def success: CheckProposeConstraintsResult = CheckProposeConstraintsSuccess
+  def success: CheckProposeConstraintsResult                    = CheckProposeConstraintsSuccess
+  def notBonded: CheckProposeConstraintsResult                  = NotBonded
+  def notEnoughNewBlock: CheckProposeConstraintsResult          = NotEnoughNewBlocks
+  def tooFarAheadOfLastFinalized: CheckProposeConstraintsResult = TooFarAheadOfLastFinalized
```

Comment:
these methods are more like renaming from my perspective. I am curious why we need these methods.

Reply:
I made this to be able to assert in tests `assert(r == ProposeResult.notEnoughBlocks && b.isEmpty)`
It does not work with case objects

Reply:
I think you are pointing to another class ProposeResult.

But the reason is not case object because the same is with case classes. It's about unifying the type, in this example each constructor has a function returning `CheckProposeConstraintsResult`.